core/src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -1056,6 +1055,26 @@ protected QuestionDef parseControl(IFormElement parent, Element e, int controlTy
         return question;
     }
 
+    private void parseControlChildren(Element e, QuestionDef question, IFormElement parent,
+                                      boolean isSelect) {
+        for (int i = 0; i < e.getChildCount(); i++) {
+            int type = e.getType(i);
+            Element child = (type == Node.ELEMENT ? e.getElement(i) : null);
+            String childName = (child != null ? child.getName() : null);

```

Comment:
if `child == null` then we should be done here, all the subsequent logic expects `childName` to be something that isn't `null`